mcos/optimizer.py

```diff
@@ -181,3 +183,100 @@ def _get_optimal_portfolio(self, cov: np.array, mu: np.array) -> np.array:
         w = np.dot(inv, mu)
         w /= np.dot(ones.T, w)
         return w.flatten()
+
+
+class HRPOptimizer(AbstractOptimizer):
+    """
+    Hierarichal Risk Parity Optimizer based on Dr. Marcos Lopez de Prado's paper 'Building Diversified Portfolios that
+     Outperform Out-of-Sample'
+    """
+
+    def allocate(self, mu: np.array, cov: np.array) -> np.array:
+        """
+       Gets position weights according to the hierarchical risk parity method as outlined in Marcos Lopez de Prado's
+       book
+       :param cov: covariance matrix
+       :return: List of position weights. Note that lru_cache has problems caching numpy arrays.
```

Comment:
Is `lru_cache` used anywhere in here? Can probably remove this comment if it's not